modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/Sidebar.js

```diff
@@ -9,31 +9,218 @@
  * distribution rights of the Software.
  */
 
-import React from 'react';
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useContext, useEffect, useState} from 'react';
 
+import {DiagramBuilderContext} from '../../DiagramBuilderContext';
 import {nodeDescription, nodeTypes} from '../nodes/utils';
+import SidebarPanel from './SidebarPanel';
 
 export default function Sidebar() {
+	const {
+		selectedNode,
+		selectedNodeNewId,
+		setSelectedNode,
+		setSelectedNodeNewId,
+	} = useContext(DiagramBuilderContext);
+	const [errors, setErrors] = useState({
+		id: false,
+		label: false,
+	});
+
 	const onDragStart = (event, nodeType) => {
 		event.dataTransfer.setData('application/reactflow', nodeType);
 		event.dataTransfer.effectAllowed = 'move';
 	};
 
+	useEffect(() => {
+		setSelectedNodeNewId(null);
+		setErrors({
+			id: false,
+			label: false,
+		});
+	}, [selectedNode?.id, setSelectedNodeNewId]);
+
 	return (
 		<div className="sidebar">
 			<div className="sidebar-header">
-				<span className="title">{Liferay.Language.get('nodes')}</span>
+				{selectedNode && (
```

Comment:
Hey @naraandrade, Can we break this component into smaller ones? What do you think?

Reply:
Yes, I can create a SidebarHeader component to control this visualization, as I did with the SidebarPanel. Thx :)

Reply:
If you see that this approach could be used in other parts of this file, go ahead. Thx